app/models/location.rb

```diff
@@ -0,0 +1,68 @@
+class Location
+  def initialize(city, province, country)
+    @city = city
+    @province = province
+    @country = country
+  end
+
+  attr_reader :city, :province, :country
+
+  # Listed in the order of significance.
+  # The name "component" is pulled from USPS
+  # https://pe.usps.com/text/pub28/28c2_012.htm#ep526349
+  COMPONENTS = [:city, :province, :country]
+
+  def component(compon)
+    send(compon) if compon.in? COMPONENTS
+  end
+
+  def components
+    [@city, @province, @country]
+  end
+
+  def sig_component
```

Comment:
sure, go ahead and refactor that in!

Reply:
I decided to keep "significant" since it's used in MSB (Most Significant Bit), a CS concept that represents this idea pretty well.

[`c93572c` (#73)](https://github.com/hackclub/hackathons-backend/pull/73/commits/c93572c07ab5efbbfd80d2bbcd34534247312af1)